packages/react/context-menu/src/ContextMenu.tsx

```diff
@@ -38,31 +70,79 @@ const ContextMenu: React.FC<{ onOpenChange?(open: boolean): void }> = (props) =>
   );
 
   return (
-    <MenuPrimitive.Root open={open} onOpenChange={handleOpenChange}>
+    <MenuPrimitive.Root open={open} onOpenChange={handleOpenChange} dir={dir}>
       <ContextMenuProvider open={open} onOpenChange={handleOpenChange}>
         {children}
       </ContextMenuProvider>
     </MenuPrimitive.Root>
   );
 };
 
+/* -----------------------------------------------------------------------------------------------*/
+
+type ContextMenuSubOwnProps = {
+  open?: boolean;
+  onOpenChange?(open: boolean): void;
+  defaultOpen?: boolean;
+  dir?: Direction;
+};
+
+const ContextMenuSub: React.FC<ContextMenuSubOwnProps> = (props) => {
+  const { open: openProp, defaultOpen, onOpenChange, ...menuSubProps } = props;
+  const [open = false, setOpen] = useControllableState({
+    prop: openProp,
+    defaultProp: defaultOpen,
+    onChange: onOpenChange,
+  });
+
+  return <MenuPrimitive.Sub {...menuSubProps} open={open} onOpenChange={setOpen} />;
+};
+
 ContextMenu.displayName = CONTEXT_MENU_NAME;
 
 /* -------------------------------------------------------------------------------------------------
  * ContextMenuTrigger
  * -----------------------------------------------------------------------------------------------*/
 
 const TRIGGER_NAME = 'ContextMenuTrigger';
-const TRIGGER_DEFAULT_TAG = 'span';
+const ROOT_TRIGGER_DEFAULT_TAG = 'span';
 
-type ContextMenuTriggerOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ContextMenuTriggerOwnProps = Polymorphic.Merge<
+  Polymorphic.OwnProps<typeof ContextMenuRootTrigger>,
+  Polymorphic.OwnProps<typeof MenuPrimitive.SubTrigger>
+>;
 type ContextMenuTriggerPrimitive = Polymorphic.ForwardRefComponent<
-  typeof TRIGGER_DEFAULT_TAG,
+  Polymorphic.IntrinsicElement<typeof ContextMenuRootTrigger>,
   ContextMenuTriggerOwnProps
 >;
 
 const ContextMenuTrigger = React.forwardRef((props, forwardedRef) => {
-  const { as = TRIGGER_DEFAULT_TAG, ...triggerProps } = props;
+  const { as, ...triggerProps } = props;
+  const isSubmenu = React.useContext(SubmenuContext);
+
+  return isSubmenu ? (
+    <MenuPrimitive.SubTrigger
+      {...triggerProps}
+      as={as as Polymorphic.IntrinsicElement<typeof ContextMenuRootTrigger>}
```

Comment:
Not sure on the best way to merge these two `IntrinsicElement` types so decided cast here. Is there a better approach?

Reply:
We all discussed this and decided try and address this `IntrinsicElement ` conundrum via a separate `TriggerItem` component. Will raise as a separate improvement.